### PR TITLE
09/separatepageforcategory

### PR DESCRIPTION
**Pull request by code-emad**
functionality for query by category is working
path is on /?category='insert category' 

if invalid cat inserted (e.g. http://localhost:3000/?category=notacat), it will return 'category not found'

available categories are:
strategy,
hidden-roles,
dexterity,
push-your-luck,
roll-and-write,
deck-building,
engine-building

### CODE REVIEW

File: src/Components/AllReviews.jsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
   const [reviews, setReviews] = useState([]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const category = searchParams.get("category");
 
   useEffect(() => {
-    getReviews().then((reviews) => {
+    getReviews(category).then((reviews) => {
       setReviews(reviews);
     });
   }, []);
```
Comment (Hard-Luck) [inline src/Components/AllReviews.jsx]: you need categories in the dependancy array, react is throwing a warning, these shouldn't be ignored.

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
We should have some interface in order to get the categories

File: src/Components/SingleReview.jsx [hunk 2]
```diff
@@ -15,10 +14,11 @@ export default function SingleReview() {
       getReviewById(review_id).then((review) => {
         setReview(review);
       });
-  }, [review_id]);
+  }, [review_id,  <AddComment />]);
```
Comment (MrMcSnail) [inline src/Components/SingleReview.jsx]: components aren't meant to be in a dependancy...

**Review by MrMcSnail — CHANGES_REQUESTED** [review]
Line 7 in Review Cards is not the correct url and is causing your single review page to fail. YOu also got a key warning that needs to be bashed on the head.

File: src/Components/SingleReview.jsx (hunk 2 above)
Comment (code-emad) [reply src/Components/SingleReview.jsx]: done =)

File: src/Components/AllReviews.jsx (hunk 1 above)
Comment (code-emad) [reply src/Components/AllReviews.jsx]: done =)

**Review by Hard-Luck — APPROVED** [review]

**Pull request merged**